### PR TITLE
Improve Flutter and Dart SDK using official lint rules

### PR DESCRIPTION
**Pull request by lohanidamodar**

### CODE REVIEW

File: templates/dart/lib/services/service.dart.twig [hunk 1]
```diff
@@ -38,9 +38,9 @@ class {{ service.name | caseUcfirst }} extends Service {
         };
 
 {% if method.type == 'location' %}
-        params.keys.forEach((key) {if (params[key] is int || params[key] is double) {
+        for(var key in params.keys) {if (params[key] is int || params[key] is double) {
```
Comment (wess) [inline templates/dart/lib/services/service.dart.twig:41]: Why a for loop when you could use filter/map/reduce? (Map is faster and resource efficient).

File: templates/dart/lib/src/client_mixin.dart.twig [hunk 2]
```diff
@@ -40,15 +40,15 @@ class ClientMixin {
     } else if (method == HttpMethod.get) {
       final encoded = <String, dynamic>{};
       if (params.isNotEmpty) {
-        params.keys.forEach((key) {
+        for(var key in params.keys) {
```
Comment (wess) [inline templates/dart/lib/src/client_mixin.dart.twig:43]: Why a for loop when you could use a map? (Map is faster and resource efficient).

File: templates/flutter/lib/services/service.dart.twig [hunk 3]
```diff
@@ -61,9 +61,11 @@ class {{ service.name | caseUcfirst }} extends Service {
       return client.webAuth(url);
 
 {% elseif method.type == 'location' %}
-        params.keys.forEach((key) {if (params[key] is int || params[key] is double) {
-          params[key] = params[key].toString();
-        }});
+        for (var key in params.keys) {
```
Comment (wess) [inline templates/flutter/lib/services/service.dart.twig:64]: Suggest using a map/filter here

File: templates/flutter/lib/src/realtime_mixin.dart.twig [hunk 4]
```diff
@@ -65,13 +66,13 @@ mixin RealtimeMixin {
             break;
           case 'event':
             final message = RealtimeMessage.fromMap(data.data);
-            message.channels.forEach((channel) {
-              if (this._channels[channel] != null) {
-                this._channels[channel]!.forEach((stream) {
+            for(var channel in message.channels) {
```
Comment (wess) [inline templates/flutter/lib/src/realtime_mixin.dart.twig:69]: since dealing with sinks, for loop looks good here.

**Review by wess — COMMENTED** [review]
Just a couple of things to consider. This looks good and will work fine, I just wanted to drop notes on some ideas going forward.

File: templates/flutter/lib/services/service.dart.twig (hunk 3 above)
Comment (lohanidamodar) [reply templates/flutter/lib/services/service.dart.twig]: removed, as I think this was a leftover

File: templates/dart/lib/src/client_mixin.dart.twig (hunk 2 above)
Comment (lohanidamodar) [reply templates/dart/lib/src/client_mixin.dart.twig]: updated

File: templates/dart/lib/services/service.dart.twig (hunk 1 above)
Comment (lohanidamodar) [reply templates/dart/lib/services/service.dart.twig]: removed, as I think this was a leftover

**Review by Meldiron — APPROVED** [review]
LGTM

**Pull request merged**